### ISSUE TITLE
Scroll to bottom when new message is sent/received doesn't always work

### ISSUE THREAD

**Issue by moezbhatti** [moezbhatti/qksms#895]

**Comment by homersimpsons** [comment]
In my mind auto scrolling to bottom should be an option, because if you are looking for an old message and you receive a new message you will have to restart your search.

A cliquable "new message(s)" should appear if the view is far from the bottom (e.g. 10 messages upper). What do you think about this?

**Comment by moezbhatti** [comment]
@homersimpsons yeah, this is exactly how I plan to implement it. Right now the app will only scroll down if you're at the very bottom message (but this doesn't always work). I plan to add a threshold to this, like 10 as you suggested, and add an indication that new messages were received if you're scrolled up too high

Any thoughts on scrolling down no matter what, if a message was sent instead of received? Or do you think they should both behave the same way?

**Comment by homersimpsons** [comment]
Okay, that's all right then :) You are doing a really good Job on QKSms, thank you.

Looking at the indication user could also have a notification (at least if he leaves without looking at the message)

> Any thoughts on scrolling down no matter what, if a message was sent instead of received? Or do you think they should both behave the same way?

I can easily think of 3 approach:
- The same behavior is all right (user have a visual confirmation that his message is in queue then sent), but the indication may mislead him if it's for example "New Message(s)", in fact the user won't know if it's his message or a received one. To avoid this we need 2 different message: "New Message(s) sent" and "New Message(s) (received?)" with the text priority for received one (wording may not be the best).
- Use short toast to say "Sending message", "Message sent" (+ receive if enable), here the user will be somehow spammed.
- Do nothing if its a success, but use a notification/indication for failed message.

I think that the user need a feedback.
